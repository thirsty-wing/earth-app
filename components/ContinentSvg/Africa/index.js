import Africa from "./Africa";


export default Africa;
