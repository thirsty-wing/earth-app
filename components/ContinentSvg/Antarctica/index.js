import Antarctica from "./Antarctica";


export default Antarctica;
