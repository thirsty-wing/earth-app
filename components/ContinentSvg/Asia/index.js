import Asia from "./Asia";


export default Asia;
