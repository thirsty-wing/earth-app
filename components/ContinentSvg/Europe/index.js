import Europe from "./Europe";


export default Europe;
