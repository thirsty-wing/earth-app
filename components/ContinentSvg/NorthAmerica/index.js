import NorthAmerica from "./NorthAmerica";


export default NorthAmerica;
