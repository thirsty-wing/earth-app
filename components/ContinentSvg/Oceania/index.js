import Oceania from "./Oceania";


export default Oceania;
