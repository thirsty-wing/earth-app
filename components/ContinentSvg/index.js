import ContinentSvg from "./ContinentSvg";


export default ContinentSvg;
