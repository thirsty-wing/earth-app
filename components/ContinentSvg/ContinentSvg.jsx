import Africa from "./Africa";
import Antarctica from "./Antarctica";
import Asia from "./Asia";
import Europe from "./Europe";
import NorthAmerica from "./NorthAmerica";
import Oceania from "./Oceania";
import SouthAmerica from "./SouthAmerica";


/**
 * Continent SVG based on code prop
 *
 * @param {object} props
 * @param {string} [props.code]
 * @returns {JSX.Element}
 */
function ContinentSvg(props) {

  const {
    code,
  } = props;

  switch(code) {
  case "AF":
    return(<Africa/>);
  case "AN":
    return(<Antarctica/>);
  case "AS":
    return(<Asia/>);
  case "EU":
    return(<Europe/>);
  case "NA":
    return(<NorthAmerica/>);
  case "OC":
    return(<Oceania/>);
  case "SA":
    return(<SouthAmerica/>);
  default:
    return(<></>);
  }

}

export default ContinentSvg;
