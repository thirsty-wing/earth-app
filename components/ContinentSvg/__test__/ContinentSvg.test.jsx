import { shallow } from "../../../utils/test/shallow";
import ContinentSvg from "../ContinentSvg";


it("should render when no code is passed in", () => {
  const content = shallow(<ContinentSvg/>);

  expect(content).toMatchSnapshot();
});

it("should render when Africa code is passed in", () => {
  const content = shallow(<ContinentSvg code="AF"/>);

  expect(content).toMatchSnapshot();
});

it("should render when Antarctica code is passed in", () => {
  const content = shallow(<ContinentSvg code="AN"/>);

  expect(content).toMatchSnapshot();
});

it("should render when Asia code is passed in", () => {
  const content = shallow(<ContinentSvg code="AS"/>);

  expect(content).toMatchSnapshot();
});

it("should render when Europe code is passed in", () => {
  const content = shallow(<ContinentSvg code="EU"/>);

  expect(content).toMatchSnapshot();
});

it("should render when North America code is passed in", () => {
  const content = shallow(<ContinentSvg code="NA"/>);

  expect(content).toMatchSnapshot();
});

it("should render when Oceania code is passed in", () => {
  const content = shallow(<ContinentSvg code="OC"/>);

  expect(content).toMatchSnapshot();
});

it("should render when South America code is passed in", () => {
  const content = shallow(<ContinentSvg code="SA"/>);

  expect(content).toMatchSnapshot();
});

