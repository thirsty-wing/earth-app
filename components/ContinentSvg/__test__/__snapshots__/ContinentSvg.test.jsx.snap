// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render when Africa code is passed in 1`] = `<Africa />`;

exports[`should render when Antarctica code is passed in 1`] = `<Antarctica />`;

exports[`should render when Asia code is passed in 1`] = `<Asia />`;

exports[`should render when Europe code is passed in 1`] = `<Europe />`;

exports[`should render when North America code is passed in 1`] = `<NorthAmerica />`;

exports[`should render when Oceania code is passed in 1`] = `<Oceania />`;

exports[`should render when South America code is passed in 1`] = `<SouthAmerica />`;

exports[`should render when no code is passed in 1`] = `<React.Fragment />`;
